StRoot/StGmtAligmentMaker/EventT.h

```diff
@@ -0,0 +1,146 @@
+  static THashList *fRotList;
+ 
+ public:
+  EventT();
+  virtual ~EventT();
+  Int_t             Build(StEvent *pEventT, Double_t pCut = 0.2);			        
+  void              Clear(Option_t *option ="");								     
+  Bool_t            IsValid() const { return fIsValid; }							     
+  static void       Reset(Option_t *option ="");								     
+  void              SetNtrack(UInt_t n) { fNtrack = n; }							     
+  void              SetNhit(UInt_t n) { fNhit = n; }							     
+  void              SetFlag(UInt_t f) { fFlag = f; }							     
+  void              SetHeader(Int_t i, Int_t run, Int_t date, Double32_t field);				     
+  TrackT           *AddTrackT();									      
+  HitT             *AddHitT();										      
+  HitT             *SetHitT(HitT *h, StHit *hit, TGeoHMatrix *comb, TrackT *track);
+  Double32_t        GetVertex(UInt_t i=0) {return (i<3)?fVertex[i]:0;}  
+  UInt_t            GetTotalNoTracks() const {return fNPTracks;}	       
+  UInt_t            GetNtrack() const { return fNtrack; }	         
+  UInt_t            GetNhit() const { return fNhit; }		       
+  UInt_t            GetFlag() const { return fFlag; }		       
+  EventTHeader     *GetHeader() { return &fEvtHdr; }                 
+  const Double32_t *GetVertex() const {return fVertex;}
+  const Double32_t *GetCovMatrix() const {return fCovariantMatrix;}
+  TClonesArray     *GetTracks() const {return fTracks;}
+  TClonesArray     *GetHits() const {return fHits;}
+  TrackT           *GetTrackT(UInt_t i=0) const {return fTracks && i < fNtrack ? (TrackT*) fTracks->At(i): 0;}
+  HitT             *GetHitT(UInt_t i=0) const {return fHits && i < fNhit ? (HitT*) fHits->At(i): 0;}
+  Int_t             GetIndexOfTrackT(const TrackT *obj) const {return fgTracks->IndexOf(obj);}
+  Int_t             GetIndexOfHitT(const HitT *obj) const {return fgHits->IndexOf(obj);}
+  static void       SetRotMatrices(THashList *Rot) {fRotList = Rot;}
+  static void       RestoreListOfRotations();
+  static THashList *RotMatrices() {return fRotList;}
+  virtual void      Print(Option_t *opt="") const;
+  ClassDef(EventT,1)  //EventT structure
+};
+
+class TBase {
+
+ public :
+   TTree          *fChain;   //!pointer to the analyzed TTree or TChain
+   Int_t           fCurrent; //!current Tree number in a TChain
+   EventT         *fEvent;
+   TString         fOutFileName;
+
+   TBase( TTree *tree = 0,
+          const Char_t *f_name = "/star/data09/calib/fisyak/Pass112/TpcSsd/065/Event_6065045_raw_1010001.root") : fEvent(0) {
+     // if parameter tree is not specified (or zero), connect the file
+     // used to generate this class and read the Tree.
+     if (tree == 0) {
+       TFile *f = (TFile*)gROOT->GetListOfFiles()->FindObject(f_name);
+       if (!f) {
+         f = new TFile(f_name);
+       }
+       tree = (TTree*)gDirectory->Get("T");
+     }
+     Init(tree);
+   }
+
+   virtual ~TBase() {if (!fChain) return; delete fChain->GetCurrentFile();}
```

Comment:
How can you be sure that the current file is the one you opened in this maker? What if another maker makes a different file current?
Also, are you sure the file is written and closed when you call it destructor?